hummingbot/connector/gateway/clob_spot/data_sources/clob_api_data_source_base.py

```diff
@@ -44,6 +45,8 @@ def __init__(
         self._forwarders_map: Dict[Tuple[Enum, Callable], EventForwarder] = {}
         self._gateway_order_tracker: Optional[GatewayOrderTracker] = None
         self._markets_info: Dict[str, Any] = {}
+        self.parent: Optional[ExchangePyBase] = None
```

Comment:
I think that this modifications are more specific to Kujira and you should inherit CLOBAPIDataSourceBase and implement your changes there